resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -252,11 +252,11 @@ else if (valueOf == null)
    }
    
 	private Class<? extends Collection> convertParameterTypeToCollectionType() {
-		if (List.class.isAssignableFrom(type)) {

```

Comment:
We should not use `isAssignableFrom(...)` method because if resource method is as follow:

> 	@GET
	public Response singleValuedQueryParam(@QueryParam("date") LinkedHashSet<Date> dates) {
		...
	}

According to the original code the collection type to instantiate will be `HashSet` and at runtime it will fail with error like: 

> 01:30:33,911 ERROR [org.jboss.resteasy.resteasy_jaxrs.i18n] (default task-3) RESTEASY002005: Failed executing GET /singleValuedQueryParam: org.jboss.resteasy.spi.InternalServerErrorException: RESTEASY003020: Bad arguments passed to public javax.ws.rs.core.Response org.jboss.resteasy.test.resource.param.resource.ParamConverterResource.singleValuedQueryParam(java.util.LinkedHashSet)  ( java.util.HashSet [Mon Dec 19 00:00:00 CET 2016, Sat Dec 17 00:00:00 CET 2016, Sun Dec 18 00:00:00 CET 2016] )
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:180)
	...
Caused by: java.lang.IllegalArgumentException: argument type mismatch
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:140)
	... 46 more

I think it's a mis understanding of the spec :

> 5.Be List, Set or SortedSet, where T satisfies 2, 3 or 4 above. The resulting collection is read-only. 

and not

> 5.Be **subtype** of List, Set or SortedSet, where T satisfies 2, 3 or 4 above. The resulting collection is read-only.